src/test/java/io/managed/services/test/cluster/DataPlaneClusterTest.java

```diff
@@ -0,0 +1,359 @@
+                Duration.ofSeconds(5),
+                Duration.ofMinutes(2),
+                ready -> {
+                    int remainingCapacity = FleetshardUtils.getCapacityRemainingUnitsFromMKAgent(oc, ManagedKafkaType.standard);
+                    log.info("remaining capacity in the cluster '{}'", remainingCapacity);
+
+                    // spawn new instance if there is a capacity for it
+                    int newlyObservedRemainingCapacity = FleetshardUtils.getCapacityRemainingUnitsFromMKAgent(oc, ManagedKafkaType.valueOf(mkType));
+                    log.debug("newly observed remaining capacity '{}'", newlyObservedRemainingCapacity);
+                    if (newlyObservedRemainingCapacity < observedRemainingCapacityInitially)
+                        return true;
+                    return false;
+                });
+
+            // get number of instances that were deleted while waiting for provisioning of new kafka instance
+            List<String> kafkaInstanceNamesAfterCreating = FleetshardUtils.listManagedKafka(oc, ManagedKafkaType.valueOf(mkType)).stream()
+                .map(e -> e.getMetadata().getName())
+                .collect(Collectors.toList());
+            List<String> deletedInstances = kafkaInstanceNamesBeforeCreating.stream().filter(e -> !kafkaInstanceNamesAfterCreating.contains(e)).collect(Collectors.toList());
+            log.debug("instances that have been deleted since begging of the test {}", deletedInstances);
+
+            int newlyObservedRemainingCapacity = FleetshardUtils.getCapacityRemainingUnitsFromMKAgent(oc, ManagedKafkaType.valueOf(mkType));
+            // remaining capacity should be lowered by 1, but as there are instances which may have been deleted by other users (which would free some space) we are not including them
+            Assert.assertTrue(newlyObservedRemainingCapacity <= observedRemainingCapacityInitially - 1  +  deletedInstances.size());
+        } catch (ApiGenericException e) {
+            // some users may not be able to create some types of instances, e.g., user with quota will not be able to create dev. instance
+            log.warn(e);
+            JSONObject jsonResponse = new JSONObject(e.getResponseBody());
+            if (KAFKAS_MGMT_21_CODE.equals(jsonResponse.get("code")))
+                throw new SkipException(String.format("user %s has no quota to create instance of type %s", Environment.PRIMARY_USERNAME, mkType));
+            else
+                throw  e;
+        } finally {
+            // cleanup of kafka instance
+            log.info("clean kafka instance with name '{}'", KAFKA_INSTANCE_NAME + "-" + mkType);
+            try {
+                KafkaMgmtApiUtils.deleteKafkaByNameIfExists(kafkaMgmtApi, KAFKA_INSTANCE_NAME + "-" + mkType);
+                log.info("wait until kafka is deleted");
+                KafkaMgmtApiUtils.waitUntilKafkaIsDeleted(kafkaMgmtApi, kafkaRequest.getId());
```

Comment:
As soon as the Kafka is deleted, we've to check that the remaining capacity in the MK AgentCR has increased by at least 1.

Reply:
so you suggest add this assertion to this test right ?

Reply:
Yes please. That will cover the two cases.

Reply:
okay excellent :)